components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/constant/SQLConstants.java

```diff
@@ -0,0 +1,122 @@
+    public static final String GET_SCOPES_BY_API_ID = "SELECT ID, NAME, DISPLAY_NAME, DESCRIPTION, API_ID, TENANT_ID "
+            + "FROM SCOPE WHERE API_ID = ? AND TENANT_ID = ?";
+    public static final String GET_API_RESOURCE_BY_IDENTIFIER = "SELECT" +
+            " AR.ID AS API_RESOURCE_ID," +
+            " AR.NAME AS API_RESOURCE_NAME," +
+            " AR.IDENTIFIER AS API_RESOURCE_IDENTIFIER," +
+            " AR.DESCRIPTION AS API_RESOURCE_DESCRIPTION," +
+            " AR.TENANT_ID AS API_RESOURCE_TENANT_ID," +
+            " AR.TYPE AS API_RESOURCE_TYPE," +
+            " AR.REQUIRES_AUTHORIZATION AS REQUIRES_AUTHORIZATION," +
+            " S.ID AS SCOPE_ID," +
+            " S.NAME AS SCOPE_QUALIFIED_NAME," +
+            " S.DISPLAY_NAME AS SCOPE_DISPLAY_NAME," +
+            " S.DESCRIPTION AS SCOPE_DESCRIPTION" +
+            " FROM API_RESOURCE AR LEFT JOIN SCOPE S ON AR.ID = S.API_ID WHERE AR.IDENTIFIER = ? AND AR.TENANT_ID = ?";
+    public static final String IS_API_RESOURCE_EXIST_BY_IDENTIFIER = "SELECT ID FROM API_RESOURCE WHERE " +
+            "IDENTIFIER = ? AND TENANT_ID = ?";
+    public static final String ADD_API_RESOURCE = "INSERT INTO API_RESOURCE (ID, TYPE, "
+            + "NAME, IDENTIFIER, DESCRIPTION, TENANT_ID, REQUIRES_AUTHORIZATION) VALUES (?, ?, ?, ?, ?, ?, ?)";
+    public static final String ADD_SCOPE = "INSERT INTO SCOPE (ID, "
+            + "NAME, DISPLAY_NAME, DESCRIPTION, API_ID, TENANT_ID) VALUES (?, ?, ?, ?, ?, ?)";
+    public static final String DELETE_API_RESOURCE = "DELETE FROM API_RESOURCE WHERE ID = ? AND TENANT_ID = ?";
+    public static final String DELETE_SCOPES_BY_API = "DELETE FROM SCOPE WHERE API_ID = ? AND TENANT_ID = ?";
+    public static final String UPDATE_API_RESOURCE = "UPDATE API_RESOURCE SET NAME = ?, DESCRIPTION = ?" +
+            " WHERE ID = ?";
+    public static final String IS_SCOPE_EXIST_BY_ID = "SELECT ID FROM SCOPE WHERE ID = ? AND TENANT_ID = ?";
+    public static final String GET_SCOPE_BY_NAME = "SELECT ID, NAME, DISPLAY_NAME, DESCRIPTION, API_ID, TENANT_ID "
+            + "FROM SCOPE WHERE NAME = ? AND TENANT_ID = ?";
+    public static final String GET_SCOPE_BY_NAME_API_ID = "SELECT ID, NAME, DISPLAY_NAME, DESCRIPTION, API_ID, " +
+            "TENANT_ID FROM SCOPE WHERE NAME = ? AND API_ID = ? AND TENANT_ID = ?";
+    public static final String GET_SCOPES_BY_TENANT_ID = "SELECT ID, NAME, DISPLAY_NAME, DESCRIPTION, API_ID, " +
+            "TENANT_ID FROM SCOPE WHERE ";
+    public static final String GET_SCOPES_BY_TENANT_ID_TAIL = " TENANT_ID = ?";
+    public static final String DELETE_SCOPE_BY_ID = "DELETE FROM SCOPE WHERE ID = ? AND TENANT_ID = ?";
+    public static final String DELETE_SCOPE_BY_NAME = "DELETE FROM SCOPE WHERE NAME = ? AND TENANT_ID = ?";
+    public static final String SEARCH_API_RESOURCE_BY_ID = "SELECT ID, NAME, IDENTIFIER, DESCRIPTION, TENANT_ID, " +
```

Comment:
check this query for sql injection possibilities

Reply:
This is no longer needed